Test adding contour lines

acalcutt opened the pull request:
This is a test I did of adding contour lines on top of your recent PR. is is nowhere near complete, but I wanted to see if you had any opinions on it. 

This uses you elevation endpoint as an example and makes one for contour lines
`'^/:id/contour/:z([0-9]+)/:x([-.0-9]+)/:y([-.0-9]+)', `

The contour endpoint generates vector tiles using a new feature of maplibre contour which allows you to overwrite the fetch tiles and decode image functions https://github.com/onthegomap/maplibre-contour/pull/359 . I have made functions to work with that feature

okimiko commented:
This looks great :)

I have a data set for contour lines at the moment, which is not used that often because its size > benefit.
If it is possible I would like to have a label for each 100m (or another useful value). Then I'm really looking forward to update my contour style!

acalcutt commented:
I rebased this and removed some extra d3-contour testing I was trying out that I didn't mean to include in this.

In the current setup it generates these line increments in the vector tiles
```
<= 8- 1000m
<= 10 - 500m
<= 11 - 250m
<= 12 - 100m
<= 13 - 50m
<= 14 -  25m
<= 15 - 20m
<= 17 - 10m
>= 18 - 5m
```
 
Since it generates vector lines though you can really set your style to show whatever you want, like this example page
https://stackblitz.com/edit/web-platform-yov93nft?file=style.json (provided the vector lines exist from tileserver). It uses https://github.com/onthegomap/maplibre-contour/ in the backgroud, so the style tips from there also somewhat apply

okimiko commented:
I had just the time to test this with my low resolution terrain file (up to zoom 7) and even that looks good so far (over zoomed). :)

Thanks for the links regarding contour-labels. I think it may be worth to add them. I will give it a try the next few days, but I'm a bit short of time at the moment.

okimiko commented:
@acalcutt: I updated the [terrain-elevation-contour](/okimiko/tileserver-gl/tree/terrain-elevation-contour) with the latest changes of the express-5 branch and applied the required changes to the contour router.

I'm not sure, if your original contour code was ready for merge, but I think the generated result looks good (may be a bit color tuning is still needed).
